concent_api/core/payments/bankster.py

```diff
@@ -75,19 +76,17 @@ def claim_deposit(
     # and also for the provider if there's a non-zero claim against his account.
     # This is done in single database transaction.
     with non_nesting_atomic(using='control'):
-        requestor_client: Client = get_or_create_with_retry(Client, public_key=requestor_public_key)
+        requestor_client = Client.objects.get_or_create_full_clean(public_key=requestor_public_key)
 
-        requestor_deposit_account: DepositAccount = get_or_create_with_retry(
-            DepositAccount,
+        requestor_deposit_account = DepositAccount.objects.get_or_create_full_clean(
             client=requestor_client,
             ethereum_address=requestor_ethereum_address,
         )
 
         if is_claim_against_provider:
-            provider_client: Client = get_or_create_with_retry(Client, public_key=provider_public_key)
+            provider_client = Client.objects.get_or_create_full_clean(public_key=provider_public_key)
 
-            provider_deposit_account: DepositAccount = get_or_create_with_retry(
-                DepositAccount,
+            provider_deposit_account = DepositAccount.objects.get_or_create_full_clean(
                 client=provider_client,
```

Comment:
I see that it's already been broken before the pull request but anyway, there's a problem here: we're not locking `DepositAccount` before doing operations on `DepositClaims`. The atomic block below **must** fetch `provider_deposit_account` and `requestor_deposit_account` again with `select_for_update()`. Otherwise any other process would be able to run `claim_deposit()` in parallel and not see the funds required by the current process as already reserved.

Reply:
Added in separate commit